packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboard.ts

```diff
@@ -147,31 +147,36 @@ export const useGridKeyboard = (
 
   const handleCellKeyDown = React.useCallback(
     (params: GridCellParams, event: React.KeyboardEvent) => {
-      if (!isGridCellRoot(event.target as HTMLElement)) {
+      // The target is not an element when triggered by a Select inside the cell
+      // See https://github.com/mui-org/material-ui/issues/10534
+      if (event.target instanceof HTMLElement && !isGridCellRoot(event.target)) {
         return;
       }
       if (event.isPropagationStopped()) {
         return;
       }
-      const isEditMode = params.cellMode === 'edit';
+
+      // Get the latest params because the cell mode may have changed by another listener
```

Comment:
Fixes a bug I found when pressing the spacebar in a editable cell in view mode.

![ivnNjiCVbw](https://user-images.githubusercontent.com/42154031/120520369-15a62a00-c3aa-11eb-837b-32256ab9edaf.gif)